confgenerator/nginx.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package confgenerator
+
+import "github.com/GoogleCloudPlatform/ops-agent/confgenerator/otel"
+
+type MetricsReceiverNginx struct {
+	ConfigComponent `yaml:",inline"`
+
+	MetricsReceiverShared `yaml:",inline"`
+
+	Endpoint string `yaml:"endpoint" validate:"omitempty,url"`
+}
+
+const defaultEndpoint = "http://localhost/status"
+
+func (r MetricsReceiverNginx) Type() string {
+	return "nginx"
+}
+
+func (r MetricsReceiverNginx) Pipelines() []otel.Pipeline {
+	if r.Endpoint == "" {
+		r.Endpoint = defaultEndpoint
+	}
+	return []otel.Pipeline{{
+		Receiver: otel.Component{
+			Type: "nginx",
+			Config: map[string]interface{}{
+				"collection_interval": r.CollectionIntervalString(),
+				"endpoint":            r.Endpoint,
+			},
+		},
+		Processors: []otel.Component{
+			otel.NormalizeSums(),
+			otel.MetricsTransform(
+				otel.AddPrefix("custom.googleapis.com"),
```

Comment:
I think this is supposed to be `workload.googleapis.com`

Reply:
Nice catch!

Reply:
I've been told that this namespace will eventually be available outside of google, and I've periodically attempted to write to it, but it still appears to be restricted:
```
2021-08-23T19:27:41.949-0400    info    exporterhelper/queued_retry.go:316      Exporting failed. Will retry the request after interval.        {"kind": "exporter", "name": "googlecloud", "error": "rpc error: code = PermissionDenied desc = User dan.jaglowski@bluemedora.com does not have permission to write to metric workloads.googleapis.com/nginx.requests.", "interval": "10.402171842s"}
```

Reply:
Should it be `workloads.googleapis.com` instead of `workloads.googleapis.com`?

Reply:
@djaglowski Did you try just "workload.googleapis.com"  It's not plural.

Reply:
For some reason I was sure I'd been told `workloads`, but checking `workload`, it works. Pushed the update.